Source/CarthageKit/Project.swift

```diff
@@ -1208,13 +1208,18 @@ public func cloneOrFetch(
 				.flatMap(.merge) { isRepository -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
 					if isRepository {
 						let fetchProducer: () -> SignalProducer<(ProjectEvent?, URL), CarthageError> = {
-							guard FetchCache.needsFetch(forURL: remoteURL) else {
+							guard FetchCache.needsFetch(forURL: repositoryURL) else {
 								return SignalProducer(value: (nil, repositoryURL))
 							}
 
 							return SignalProducer(value: (.fetching(dependency), repositoryURL))
 								.concat(
-									fetchRepository(repositoryURL, remoteURL: remoteURL, refspec: "+refs/heads/*:refs/heads/*")
+									// Don't need to pass `remoteURL` here since the repository should have 
+									// the `origin` remote.
+									//
+									// See https://github.com/Carthage/Carthage/issues/968
+									// and https://github.com/Carthage/Carthage/pull/2125.
+									fetchRepository(repositoryURL, refspec: "+refs/heads/*:refs/heads/*")
```

Comment:
The only other place we use this function is when we fetch submodules, which we have also set up with the 'origin' remote. It might it make sense to just remove the remoteURL argument and fix up the submodule usage as well.